cabal install cabal-install: parse error on input ‘-- $ caba

I was running 
```
PATH=/opt/ghc/ghc-8.4.4/bin:$PATH cabal install cabal-install
```
with `ghc-options=-haddock` in `~/.cabal/config`. Result:
```
[150 of 150] Compiling Main             ( main/Main.hs, dist/build/cabal/cabal-tmp/Main.o )

main/Main.hs:596:5: error:
    parse error on input ‘-- $ cabal sandbox init && cabal configure -w /path/to/ghc
...
Failed to install cabal-install-2.4.1.0
```

--- a/cabal-install/main/Main.hs
+++ b/cabal-install/main/Main.hs
@@ -585,7 +585,7 @@ installAction
     -- TODO: It'd be nice if 'cabal install' picked up the '-w' flag passed to
     -- 'configure' when run inside a sandbox.  Right now, running
     --
-    -- $ cabal sandbox init && cabal configure -w /path/to/ghc
+    -- \$ cabal sandbox init && cabal configure -w /path/to/ghc
     --   && cabal build && cabal install
     --
     -- performs the compilation twice unless you also pass -w to 'install'.

